tcms/report/data.py

```diff
@@ -310,38 +328,47 @@ def _get_builds(self):
             self.report_criteria
         )
 
-    def automation_total(self, build_ids=[]):
-        automated_name_map = {}
+    @staticmethod
+    def automation_total(build_ids):
+
+        test_case_runs = TestCaseRun.objects.filter(
+            build__in=build_ids
+        ).values(
+            'case__is_automated'
+        ).distinct().annotate(total_count=Count('case__is_automated'))
 
-        # convert tuples to map for easier indexing
-        for _id, name in AUTOMATED_SERCH_CHOICES:
-            if isinstance(_id, int):
-                automated_name_map[_id] = name
+        test_case_runs = TestCaseRun.objects.filter(
+            build__in=build_ids
+        ).values(
+            'case__is_automated'
+        ).distinct().annotate(total_count=Count('case__is_automated'))
 
         results = {}
-        for obj in TestCaseRun.objects.filter(
-                build__in=build_ids
-            ).values('case__is_automated').distinct().annotate(
-                total_count=Count('case__is_automated')):
-            automation_id = obj['case__is_automated']
-            results[automated_name_map[automation_id]] = obj['total_count']
+        for test_case_run in test_case_runs:
+            automation_id = test_case_run['case__is_automated']
+            results[AUTOMATED_SERCH_CHOICES_MAP[automation_id]] = test_case_run['total_count']
```

Comment:
this map is duplicating content from the AUTOMATED_SERCH_CHOICES tuple and is used in only one place. If you'd like to use a dict for easier indexing then construct the dict inside the body of this method.